src/components/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react';
+
+const ThemeSwitcher = () => {
+  const [theme, setTheme] = useState<any>('');
```

Comment:
We can use `useState` lazy initialiser to read the `theme` value from `localStorage`. This is better than doing it in `useEffect` as it prevents an extra re-render(and possible flash of theme change).
Also use correct types when known instead of using `any`.

```suggestion
  const [theme, setTheme] = useState<'light' | 'dark'>(() => localStorage.getItem('theme') ?? 'light');
```

Reply:
Flash is happening here too. 😔

Reply:
I see. We should run theme switching script in the head section of the html to resolve the flashing. Also, I noticed this is the cause for the dark mode icon not showing up when theme is fetched from localStorage(hydration issues from mismatches)
Let's revert the change here then.